### PR TITLE
pullRequestFirst

### PR DESCRIPTION
**Pull request by VitalinaZhdanko**

### CODE REVIEW

File: cypress/all-tests/api-tests/testsForPet/createPet.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    })
+    it('Negative: boundary value for "category" field C24', () => {
+        let requestData=boundaryValue(2);
+        createPet(requestData, false).then(response => {
+            expect(response.status).to.eq(400);
+            expect(response.message).to.eq('Length must be between min and max');
+        })
+    })
+    it('Negative: boundary value for "photoUrls" field C25', () => {
+        let requestData=boundaryValue(3);
```
Comment (kateyurasova) [inline cypress/all-tests/api-tests/testsForPet/createPet.js]: лучше оставить формирование набора данных в самом тесте. это более наглядно. 
использование чисел для параметров не лучшее решение.

File: cypress/all-tests/api-tests/testsForPet/createPet.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        createPet(requestData, false).then(response => {
+            expect(response.status).to.eq(400);
+            expect(response.message).to.eq('Length must be between min and max');
+        })
+    })
+    it('Negative: boundary value for "tags" field C26', () => {
+        let requestData=boundaryValue(4);
+        createPet(requestData, false).then(response => {
+            expect(response.status).to.eq(400);
+            expect(response.message).to.eq('Length must be between min and max');
```
Comment (kateyurasova) [inline cypress/all-tests/api-tests/testsForPet/createPet.js]: Структура ответа немного более сложная чем тут описано
Мы ожидаем массив из javascript объектов.

 "messages": [
    {
      "fieldName": "@fieldName",
      "fieldError": "length must be between @minLength and @maxLength"
    }
  ]

тут подойдет deep equal

Comment (kateyurasova) [reply cypress/all-tests/api-tests/testsForPet/createPet.js]: messages[0].fieldName  должен быть равен name
messages[0].fieldError должен быть равен "length must be between @minlength and @Maxlength"
где @minlength and @Maxlength это значения из limits

File: cypress/all-tests/api-tests/testsForPet/createPet.js [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+                expect(response.body.id).to.be.greaterThan(0);
+                expect(response.body).to.have.property('name', pet.name);
+                expect(response.body.photoUrls).to.deep.equal(pet.photoUrls);
+                expect(response.body.tags).to.deep.equal(pet.tags);
+            })
+        })
+    });
+
+    it('Positive: Create new Pet: Chinese language in request C5', () => {
+        cy.fixture('petChinese').then(pet => {
```
Comment (kateyurasova) [inline cypress/all-tests/api-tests/testsForPet/createPet.js]: я в целом не против что ты используешь фиктуру, но можно использовать и Chance().string
Чтобы он нагенерил тебе строку из китайских символов, ты можешь использовать параметр pool
Chance().string({ pool: '广西壮族自治区少数民族古籍整理出版规划领导小组' });

File: cypress/all-tests/api-tests/testsForPet/createPet.js [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        })
+    })
+
+    it('Negative: boundary value for "name" field C23', () => {
+       let dataSet=getPetRequestData(PET_LIMIT, true);
+        dataSet.name= Chance().string({length: PET_LIMIT.name.max + 1});
+        createPet(dataSet, false).then(response => {
+            expect(response.status).to.eq(400);
+            expect(response.message).to.deep.eq('fieldName','@fieldName');
+            expect(response.message).to.deep.eq('fieldError','Length must be between @minLength and @maxLength');
```
Comment (kateyurasova) [inline cypress/all-tests/api-tests/testsForPet/createPet.js]: response.messages[0].fieldError сравнить с 'Length must be between @minLength and @maxLength' где @minLength and @maxLength это значения 1 и 100 нужно взять их limits и подставить в эту строку

Comment (kateyurasova) [reply cypress/all-tests/api-tests/testsForPet/createPet.js]: deep.eq в данном случае делать бессмысленно. вероятно, ты не понимаешь как он работает.
Он применяется для сравнения массивов и объектов. Обычное equal для массивов и объектов не работает потому что сравнивает не сами свойства и их значений, а ссылки на объекты в памяти.

**Comment by kateyurasova** [comment]
Виталина, спасибо за правки. Можно мержить.
